core/src/main/java/hudson/search/Search.java

```diff
@@ -76,7 +76,7 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp) throws IOException,
                     SuggestedItem target = find(index, query, smo);
                     if(target!=null) {
                         // found
-                        rsp.sendRedirect2(a.getUrl()+target.getUrl());
+                        rsp.sendRedirect2(req.getContextPath()+target.getUrl());
```

Comment:
Seems like this change should be testable. `JenkinsRule` sets a nonempty context path (`/jenkins`).